server/gateway/src/odspUtils.ts

```diff
@@ -36,11 +41,73 @@ export function isSpoServer(server: string) {
     return false;
 }
 
+// TODO: These functions are taken from @fluidframework/odsp-utils package and should be removed
+// once the updated getDriveItemByRootFileName function in PR #3826 is available on public feeds
+function toIODSPDriveItem(parsedDriveItemBody: any): IOdspDriveItem {
+    const path = parsedDriveItemBody.parentReference.path !== undefined ?
+        parsedDriveItemBody.parentReference.path.split("root:")[1] : "/";
+    return {
+        path,
+        name: parsedDriveItemBody.name,
+        drive: parsedDriveItemBody.parentReference.driveId,
+        item: parsedDriveItemBody.id,
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        isFolder: !!parsedDriveItemBody.folder,
+    };
+}
+
+async function getDriveItem(
+    getDriveItemUrl: string,
+    authRequestInfo: IOdspAuthRequestInfo,
+    create: boolean,
+): Promise<IOdspDriveItem> {
+    let getDriveItemResult = await getAsync(getDriveItemUrl, authRequestInfo);
```

Comment:
Do we want to prevent the create scenario where the item already exists?

Reply:
This was straight taken from odsp-utils package and I believe that check does happen in the code since if gateway tries to fetch a file with an existing file's name, it just loads the old file